### ISSUE TITLE
 Feature request: Open bookmark in [container] (talk to the (integrate the contextualIdentities api)

### ISSUE THREAD

**Issue by v00p** [aaFn/Bookmark-search-plus-2#32]
Being able to Open bookmark in any specified container has been made possible since march 13th of this years. The official Mozilla Firefox Container addons itself haven't yet integrated this capability however. 
https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/contextualIdentities

**Comment by aaFn** [comment]
Hello @v00p , interesting, I didn't know that. However, there is a problem:

```
From Firefox 57 onwards, if an extension that uses the contextualIdentities API is installed,
then the contextual identities feature will be enabled automatically.
```

This means that if I include the `"contextualIdentities"` permission in my add-on to be able to propose the option in the contextual menu, it will force the add-on users to have this function activated, and exposed to it without any preparation.
And maybe force them to specifically disable it if they do not want it.

That is clearly an undesirable effect. Too bad they made that choice in FF API design.
And I have no way to conditionally activate that API depending on whether the user already activated it or not, since this is a static setting in the manifest.json file .. if there, it activates the feature.

Therefore, we should wait for the functionality to be present officially in normal FF builds.

Meanwhile, I suggest that people who want to use it simply open a new Window or new tab using the File menu as explained in https://wiki.mozilla.org/Security/Contextual_Identity_Project/Containers, and then use the add-on within it.
Not ideal, but should be functional .. to live with until FF officially expose the function to all its user base.

Let me know your thoughts.

**Comment by v00p** [comment]
@aaFn I agree. And this feature is so rarely updated, I don't expect them to add this functionnality anytime soon..  One way would be to branch your current extension in two versions. The a non "MUA-ready" one (current version), and a (new) "MUA-ready" one.. 
I can't figure much any way/workaround for any extension developpers to integrate this contextual identity api, without recurring to such version branching annoyance..

I wonder if the [browser.contextualIdentities.query](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/contextualIdentities/query) function alone require that "contextualIdentities" permission ...

**Comment by v00p** [comment]
Hi @aaFn 
I'm not 100% sure of that, but I believe that even the current container feature of the browser doesn't handle that on it's own. Containers in FF can be run with or without a container management extension. 
There is currently only one such extension that I know of, and it's [Mozilla Multi-Account Container](https://github.com/mozilla/multi-account-containers) .. I think this is that extension alone that handle the redirection to proper Container

**Comment by april4** [comment]
Very expected this feature got implemented.

**Comment by aaFn** [comment]
Hello @april4 , ok. This is still not yet mainstream in FF, so still waiting for FF to formally publish the function to enable it .. unless I can find a way to not force it to activate when the user did not activate it by himself.

**Comment by aaFn** [comment]
Hello @v00p , @april4 , this feature is now mainstream in FF and publicized more widely since about a month https://support.mozilla.org/en-US/kb/containers

So I got a look into it again.
Good news ! Since BSP2 2.0.69 on 01/01/2020, I have been using FF integrated context menus for bookmarks, available in FF >= 64.0, and the Mozilla [Multi-Account Containers extension](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/) also uses that, once you enable its option to `Enable Bookmark Menus` 😄.

So I have nothing more to do, the simple fact that I implemented FF integrated context menus is doing all that you want !

Let me know if you see something else. If not, I will close this issue as done (indirectly) 🙂
